widget/embedded/src/events.ts

```diff
@@ -0,0 +1,24 @@
+import type { QuoteEventData, WalletEventData } from './types';
+import type {
+  RouteEventData,
+  StepEventData,
+} from '@rango-dev/queue-manager-rango-preset';
+
+import { MainEvents as QueueManagerEvents } from '@rango-dev/queue-manager-rango-preset';
+import mitt from 'mitt';
+
+export enum MainEvents {
+  RouteEvent = QueueManagerEvents.RouteEvent,
+  StepEvent = QueueManagerEvents.StepEvent,
+  QuoteEvent = 'quoteEvent',
+  WalletEvent = 'walletEvent',
+}
+
+export type Events = {
+  [MainEvents.RouteEvent]: RouteEventData;
+  [MainEvents.StepEvent]: StepEventData;
+  [MainEvents.QuoteEvent]: QuoteEventData;
+  [MainEvents.WalletEvent]: WalletEventData;
+};
+
+export const eventEmitter = mitt<Events>();
```

Comment:
I recommend moving the event emitter to the services directory and the event types to the types directory.

Reply:
Done